lib/Backends/Backend.cpp

```diff
@@ -15,23 +15,75 @@
  */
 
 #include "glow/Backends/Backend.h"
+#include "glow/Converter/TypeAToTypeBFunctionConverter.h"
 #include "glow/Graph/Graph.h"
 #include "glow/Graph/PlaceholderBindings.h"
 #include "glow/IR/Instrs.h"
 #include "glow/Optimizer/Optimizer.h"
 
 using namespace glow;
 
-void Backend::optimizeFunction(Function *F,
-                               const CompilationOptions &opts) const {
+static void transformForPrecisionMode(const Backend &B, Function *&F,
```

Comment:
Why do we need to pass the backend around? This really complicates things. We should not continue to increase the number of parematers to all of our optimizations functions. This is a sign that we don't have clean APIs.

Reply:
The Backend is required in `optimizeFunction()` (and then into here) because it reports on what the specific Backend supports + what should be lowered (via `isOpSupported()` + `shouldLower()`), and it also allows the Backend to transform the graph (via `transformPostLowering()`).

It's passed in here because the quantizer only quantizes Nodes given what the Backend supports (via `isOpSupported()`). It's just a helper method here for `optimizeFunction()`.

One way we could consider redesigning this would be to refactor Backend into multiple classes, like a BackendPreferences class. But I don't see a way to not pass at least something for the backend in here right now.